client/rest-high-level/src/test/java/org/elasticsearch/client/EqlIT.java

```diff
@@ -61,4 +63,31 @@ public void testBasicSearch() throws Exception {
         assertNotNull(response.hits().events());
         assertThat(response.hits().events().size(), equalTo(1));
     }
+
+    public void testLargeMapping() throws Exception {
+        Request doc1 = new Request(HttpPut.METHOD_NAME, "/index/_doc/1");
+
+        String now = DateUtils.nowWithMillisResolution().format(DateTimeFormatter.ISO_DATE_TIME);
+        StringBuilder sb = new StringBuilder();
+        sb.append("{");
+        for (int i = 0; i < 250; i++) {
```

Comment:
I would add a comment here explaining the reason for 250 value (that value larger than the 100 doc_values limit in ES). Maybe take the default limit in ES and use here a value made of that limit + 100 or something like that?